basalt/__init__.mojo

```diff
@@ -1,6 +1,10 @@
 from .autograd import Graph, Symbol, OP
 from .nn import Tensor, TensorShape
+from basalt.utils.collection import Collection
 
 alias dtype = DType.float32
 alias nelts = 2 * simdwidthof[dtype]()
 alias seed = 42
+
+var TENSORS = Collection()
+var GRADS = Collection()
```

Comment:
Why was it necessary to move the tensor grads from the model to global objects?

Reply:
Another note on this, if we are using Collection for Tensors and Grads, how much performance overhead is caused by the Collection struct? I actually don't mind these being global but I do think we should make sure Collection is VERY optimized if we go this route.